### PR TITLE
everything lakefs docker compose file 

### PR DESCRIPTION
**Pull request by nopcoder**
A docker-compose file with everything we can throw into work with lakeFS.
The idea behind it is to have a quick way to demo different lakeFS integrations.

### CODE REVIEW

File: deployments/compose/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,142 @@
+version: "3"
+services:
+
```
Comment (guy-har) [inline deployments/compose/docker-compose.yml:3]: It's not everything without spark :)

**Review by guy-har — CHANGES_REQUESTED** [review]
Looks good,
Thanks.
Please add the spark containers to the docker-compose as well.

File: deployments/compose/docker-compose.yml (hunk 1 above)
Comment (nopcoder) [reply deployments/compose/docker-compose.yml:3]: Spark!

Comment (nopcoder) [reply deployments/compose/docker-compose.yml:3]: Added spark with 3 workers - note I didn't configure it to make use of lakeFS as a gateway or the underlying storage to be MINIO.

File: deployments/compose/docker-compose.yml [hunk 2]
```diff
@@ -137,6 +137,52 @@ services:
...
+    deploy:
+      replicas: 3
+    links:
+      - lakefs:s3.docker.lakefs.io
+
+  spark-submit:
+    image: docker.io/bitnami/spark:3
+    profiles: ["client"]
+    volumes:
+      - ./:/local
```
Comment (guy-har) [inline deployments/compose/docker-compose.yml:180]:
```suggestion
    volumes:
      - ./targets:/targets
```

Comment (guy-har) [reply deployments/compose/docker-compose.yml:180]: Believe this is to support spark-submit from a local target, Not sure targets is the right name, but It should be in some subdirectory.

**Review by guy-har — APPROVED** [review]

File: deployments/compose/hive/wait-for [hunk 3]
```diff
@@ -0,0 +1,79 @@
+#!/bin/sh
```
Comment (itaiad200) [inline deployments/compose/hive/wait-for:1]: We have a `wait-for` script in this repo, can we reuse it?

Comment (nopcoder) [reply deployments/compose/hive/wait-for:1]: I'll find an alternative and remove this copy of wait-for script

File: deployments/compose/docker-compose.yml (hunk 2 above)
Comment (nopcoder) [reply deployments/compose/docker-compose.yml:180]: for general use the `local` mount will use to access any file in your env.

**Pull request merged**